src/mailbox_safe.cpp

```diff
@@ -59,16 +60,8 @@ void zmq::mailbox_safe_t::add_signaler (signaler_t* signaler)
 
 void zmq::mailbox_safe_t::remove_signaler (signaler_t* signaler)
 {
-    std::vector<signaler_t*>::iterator it = signalers.begin();
-
     // TODO: make a copy of array and signal outside the lock
-    for (; it != signalers.end(); ++it){
-        if (*it == signaler)
-           break;
-    }
-
-    if (it != signalers.end())
-        signalers.erase(it);
+    signalers.erase(std::remove(signalers.begin(), signalers.end(), signaler), signalers.end());
```

Comment:
This will break on Solaris, AIX and the like as it needs newer C++ standard library